build-tools/packages/build-cli/src/commands/typetests.ts

```diff
@@ -153,29 +158,62 @@ export function previousVersion(version: string): string {
 
 /**
  * Updates configuration for type tests in package.json
+ *
+ * @internal
  */
-function updateTypeTestConfiguration(pkgJson: PackageJson, options: TypeTestConfigActions): void {
+export function updateTypeTestConfiguration(
+	pkgJson: PackageJson,
+	options: TypeTestConfigActions,
+): void {
 	if (options.version !== undefined) {
-		const oldDepName = `${pkgJson.name}-previous`;
+		applyTypeTestVersionOptions(pkgJson, options.version);
+	}
 
-		// Packages can explicitly opt out of type tests by setting typeValidation.disabled to true.
-		const enabled = pkgJson.typeValidation?.disabled !== true;
+	if (options.resetBroken !== undefined) {
```

Comment:
since TypeTestConfigActions has the property that undefined members mean don't do anything, we could make resetBroken `resetBroken?: true` instead of `boolean`. This would eliminate one case from being possible to express, and thus replace a case we might want to test with one the compiler can remove for us.

It also makes the semantics of resetBrokenTests better align with its name, since it would then always reset them.

Reply:
This was new to me. In retrospect it's obvious but it never occurred to me to define a `true | undefined` type! One downside is that this usage doesn't work any more:

```ts
updateTypeTestConfiguration(json, { resetBroken: this.flags.reset, version });
```

I can massage the flag before calling the function, but is that better?

Reply:
With an optional boolean we have 4 ways the flag could be setup:
{x: true}
{x: false}
{x: undefined}
{}

With optional true, we have three:
{x: true}
{x: false}
{x: undefined}
{}

TypeScript sometimes sorta treats the last two differently (and JS clearly treats them differently). If we prefer the last one we can do the conversion inline as:
`{ ...(this.flags.reset? {resetBroken: true}, {}), version }`
If we prefer the second:
`{ resetBroken: this.flags.reset? true : undefined, version }`

A better alternative to this might be to just delete the updateTypeTestConfiguration function and the TypeTestConfigActions type and just repalce:

```
updateTypeTestConfiguration(json, { resetBroken: this.flags.reset, version });
```

with 

```typescript
	if (version !== undefined) {
		applyTypeTestVersionOptions(pkgJson, version);
	}

	if (this.flags.reset) {
		resetBrokenTests(pkgJson);
	}
```

We can just delete the whole mid level abstraction.
That moves slightly more logic into processPackage which is less testable, but deletes enough concepts that its likley a win.

If we ever to want to test all the actual flag interpertation logic, we could hoist the whole update json callback from processPackage out into a named function that takes the flags object and give it a basic integration test.

Reply:
Thats a specific application of the general approach of "if a data conversion seems like it would be nice, try and just delete enough abstractions/layers so the conversion isn't needed in the first place approach".

Reply:
I removed the whole abstraction. Like you said, it wasn't adding anything.